SCSH_3C_iOS/SCSH_3C_iOS/Modules/Home/View/BodyCollectionViewCell.swift

```diff
@@ -0,0 +1,150 @@
+            transparentButtonsHeaderStackView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            transparentButtonsHeaderStackView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+            transparentButtonsHeaderStackView.heightAnchor.constraint(equalToConstant: 50),
+            
+            previousButton.widthAnchor.constraint(equalTo: nextButton.widthAnchor),
+            previousButton.widthAnchor.constraint(equalTo: centerButton.widthAnchor, multiplier: 0.5)
+        ])
+    }
+
+    private func setupBodyLayout() {
+        contentView.addSubview(bodyView)
+        
+        NSLayoutConstraint.activate([
+            bodyView.topAnchor.constraint(equalTo: transparentButtonsHeaderStackView.bottomAnchor),
+            bodyView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            bodyView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+            bodyView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor)
+        ])
+    }
+    
+    private func setupAction() {
+        previousButton.addTarget(self, action: #selector(previousButtonDidTap), for: .touchUpInside)
+        centerButton.addTarget(self, action: #selector(centerButtonDidTap), for: .touchUpInside)
+        nextButton.addTarget(self, action: #selector(nextButtonDidTap), for: .touchUpInside)
+    }
+    
+    @objc
+    private func previousButtonDidTap() {
+        guard let currentIndexPath = currentIndexPath else { return }
+        let previousIndexPath = IndexPath(row: currentIndexPath.row - 1, section: currentIndexPath.section)
+        
+        delegate?.scrollBodyCollectionView(at: previousIndexPath, animated: true)
+    }
+    
+    @objc
+    private func centerButtonDidTap() {
+        guard let currentIndexPath = currentIndexPath else { return }
+        
+        delegate?.scrollBodyCollectionView(at: currentIndexPath, animated: true)
+    }
+    
+    @objc
+    private func nextButtonDidTap() {
+        guard let currentIndexPath = currentIndexPath else { return }
+        let nextIndexPath = IndexPath(row: currentIndexPath.row + 1, section: currentIndexPath.section)
+        
+        delegate?.scrollBodyCollectionView(at: nextIndexPath, animated: true)
+    }
+    
+    func setup(embeddedviewController: UIViewController, to parentViewController: UIViewController, indexPath: IndexPath) {
+        setup(indexPath: indexPath)
+        
+        // Store the reference to the view controller
+        self.embeddedViewController = embeddedviewController
+
+        // Add the view controller as a child view controller
+        parentViewController.addChild(embeddedviewController)
+
+        // Set the frame of the view controller's view
+        embeddedviewController.view.frame = self.bodyView.bounds
+
+        // Add the view controller's view to the cell's content view
+        self.bodyView.addSubview(embeddedviewController.view)
```

Comment:
I believe that the comments here are not needed. The code appears to be self-explanatory.

There are some comments in other parts of the code that may not be needed as well.

Reply:
absolutely agree, I forgot to delete it 😅

Reply:
[done](https://github.com/No-57/SCSH_3C_iOS/pull/27/commits/30ae5ec3e8ab3cafaa1fa0be661683db15d23d4e)